### PR TITLE
[NVIDIA XLA] Add a new debug flag to turn off mlir pretty print form

### PR DESCRIPTION
**Pull request by Tixxx**
Introduce a debug flag "xla_dump_enable_mlir_pretty_form" to control pretty print form when dumping mlir. 
Disabling pretty form will make the dumped mlir parsable. LLVM states the pretty form is not parsable by any optimizer tool.
With pretty form:
`func.func @main.3(%arg0: memref<4xi8> {lmhlo.params = 0 : index} "Arg_0.1"`
The string at the end is not parsable
With `xla_dump_enable_mlir_pretty_form=false`:
`func.func @main.3(%arg0: memref<4xi8> {lmhlo.params = 0 : index} loc("Arg_0.1")`

### CODE REVIEW

**Comment by cheshire** [comment]
@akuegel I think this was merged manually?

File: tensorflow/compiler/xla/debug_options_flags.cc [hunk 1]
```diff
@@ -50,6 +50,7 @@ DebugOptions DefaultDebugOptionsIgnoringFlags() {
   opts.set_xla_dump_max_hlo_modules(-1);
   opts.set_xla_dump_module_metadata(false);
   opts.set_xla_dump_hlo_as_long_text(false);
+  opts.set_xla_dump_disable_mlir_pretty_form(false);
```
Comment (akuegel) [inline tensorflow/compiler/xla/debug_options_flags.cc:53]: I think it is easier to reason about without the double negation here.
Why not name the flag set_xla_dump_enable_mlir_pretty_form and set it to true by default?

**Review by akuegel — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/debug_options_flags.cc (hunk 1 above)
Comment (Tixxx) [reply tensorflow/compiler/xla/debug_options_flags.cc]: sounds good to me, changed it to "enable"

**Comment by Tixxx** [comment]
hi @akuegel , could you take another look please? Thanks.

**Review by akuegel — APPROVED** [review]

**Review by akuegel — APPROVED** [review]

**Pull request merged**